### PR TITLE
gh-107559: Argument Clinic: complain about non-ascii chars in param docstrings

### PR DESCRIPTION
**Pull request by erlend-aasland**
Also, improve the warn() message.

* Issue: gh-107559

### CODE REVIEW

File: Lib/test/test_clinic.py [hunk 1]
```diff
@@ -1427,6 +1427,26 @@ def test_scaffolding(self):
...
+            module test
+            test.fn
+                a: int
+                    á param docstring
+            docstring fü bár baß
+        """
+        with support.captured_stdout() as stdout:
+            self.parse(block)
+        # Only the first match is shown.
+        # Also, the line numbers are off; this is a known limitation.
```
Comment (erlend-aasland) [inline Lib/test/test_clinic.py]: Let's try to address that limitation in the error handling rewrite.

**Review by AlexWaygood — APPROVED** [review]

**Comment by AlexWaygood** [comment]
> Do you want to backport this?

I'll leave it to you — it's strictly speaking a bugfix, so we could, but it's also a bug that people are pretty unlikely to hit in practice (and I don't know how many people actually use clinic from older branches of CPython anyway...)

File: Tools/clinic/clinic.py [hunk 2]
```diff
@@ -5266,6 +5263,11 @@ def state_parameter_docstring_start(self, line: str) -> None:
 
     def docstring_append(self, obj: Function | Parameter, line: str) -> None:
         """Add a rstripped line to the current docstring."""
+        if match := re.search(r'[^\x00-\x7F]', line):
```
Comment (AlexWaygood) [inline Tools/clinic/clinic.py]: (Optional:) You could use `re.finditer` if you wanted to iterate over all matches

Comment (erlend-aasland) [reply Tools/clinic/clinic.py]: Yes, that's even better.

Comment (erlend-aasland) [reply Tools/clinic/clinic.py]: 00b84e78cc7cc848ffc067281417b7a47ede11c7

**Comment by erlend-aasland** [comment]
> I'll leave it to you to decide — it's strictly speaking a bugfix, so we could, but it's also a bug that people are pretty unlikely to hit in practice (and I don't know how many people actually use clinic from older branches of CPython anyway...). So maybe not worth the churn.

We'll see if 🧙‍♀️ can handle it. If she can't, we won't bother.

**Review by AlexWaygood — APPROVED** [review]

**Pull request merged**